### PR TITLE
Fail if expires is missing and expiry check is turned on

### PR DESCRIPTION
**Pull request by nijel**
Skipping the verification if expires is missing brings false expectation that expiry has been checked.

### CODE REVIEW

**Comment by ovx** [comment]
Hi, thanks for your PRs! But this one would break how the library works across other languages. The check should be performed only if the salt contains the expires parameter so that the verification succeeds even with `check_expires=True` if the challenge was generated without expiration.

In terms of security, this works just fine because the expires parameter is in the salt, which is signed. So if you generate a challenge with expires and someone just removes that parameter, the signature is no longer valid and the check fails on that.

**Comment by nijel** [comment]
I think it would be better to change the behavior for other languages as well because it makes the omission of the `expires` parameter in the challenge obvious from the beginning instead of silently hiding it and letting the user think that the expiry is checked while it is not.

I created this PR because I did follow https://github.com/altcha-org/altcha-lib-py?tab=readme-ov-file#usage what doesn't set the `expires` parameter, but passes `check_expires=True` while validating. The presence of the `check_expires` made me believe that there is some default expiry and that is being checked. By reading the code, I've learned that it doesn't.

On the other side, I understand that this would be a breaking change for everybody who just copied the example (as I did), so it should be communicated as such.

**Comment by ovx** [comment]
When provided with `check_expires=True` the verification checks the expires only if the expires parameter is present in the challenge, which is correct, nothing is being silently ignored. The challenge generation is under the control of the server and the server decides whether to add the expires parameter or not and as mentioned earlier, the parameter is signed, thus if an attacker decides to remove the expires hoping that they can reuse the challenge even is you force check_expires=True, the verification would simply fail on the incorrect signature.

Simply put, the current behavior is safe, it works as expected. I updated the readme to better describe the current behavior and also added expires to the example.
